cmake/modules/AddSwift.cmake

```diff
@@ -97,8 +106,8 @@ function(_add_host_variant_c_compile_link_flags name)
     set(DEPLOYMENT_VERSION "${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_DEPLOYMENT_VERSION}")
   endif()
 
-  # MSVC, clang-cl, gcc don't understand -target.
-  if(CMAKE_C_COMPILER_ID MATCHES "Clang" AND NOT SWIFT_COMPILER_IS_MSVC_LIKE)
+  does_compiler_support_target_triple(COMPILER_SUPPORTS_TARGET_TRIPLE)
```

Comment:
The out parameters really make things more complicated, is there a reason for that change?  Im slowly working towards making this more declarative, I think I would rather have the check duplicated.

Reply:
I saw this is the pattern used for predicates (e.g. `is_darwin_based_sdk`) and I preferred to stick with that -- after a bit of experimentation, I could not see a better way to factor this kind of logic in CMake (short of declaring a global variable), but I could be missing something obvious.

Reply:
I was suggesting that we duplicate the check instead.  That makes it easier to convert this pattern to a more declarative pattern.

Reply:
Sorry, I misunderstood your comment -- check duplicated in 05af089cf0a155ba42a8c6e80f43b922defce8f5